From c57773d1ec6317b3c6eb958592c1874d2c5a8211 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Thu, 27 Apr 2023 13:18:19 +0800
Subject: [PATCH 62/68] Revert "drm: Fix potential null-ptr-deref due to
 drmm_mode_config_init()"

This reverts commit 5ae70041a6d7de62a0cdb2bbcfe0c9cf753035d0.
---
 drivers/gpu/drm/drm_mode_config.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/drm_mode_config.c b/drivers/gpu/drm/drm_mode_config.c
index fad2c11811270..f1affc1bb6799 100644
--- a/drivers/gpu/drm/drm_mode_config.c
+++ b/drivers/gpu/drm/drm_mode_config.c
@@ -398,8 +398,6 @@ static void drm_mode_config_init_release(struct drm_device *dev, void *ptr)
  */
 int drmm_mode_config_init(struct drm_device *dev)
 {
-	int ret;
-
 	mutex_init(&dev->mode_config.mutex);
 	drm_modeset_lock_init(&dev->mode_config.connection_mutex);
 	mutex_init(&dev->mode_config.idr_mutex);
@@ -421,11 +419,7 @@ int drmm_mode_config_init(struct drm_device *dev)
 	init_llist_head(&dev->mode_config.connector_free_list);
 	INIT_WORK(&dev->mode_config.connector_free_work, drm_connector_free_work_fn);
 
-	ret = drm_mode_create_standard_properties(dev);
-	if (ret) {
-		drm_mode_config_cleanup(dev);
-		return ret;
-	}
+	drm_mode_create_standard_properties(dev);
 
 	/* Just to be sure */
 	dev->mode_config.num_fb = 0;
-- 
2.25.1

